src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -58,13 +58,15 @@ def _get_transaction_id(payload):
             raise ValueError("transaction_id not in payload or None")
 
     def _register(self, payload):
-        self._connected = register_area(self._redis_communicator, self.channel_prefix,
-                                        self.connected, self._get_transaction_id(payload),
-                                        area_uuid=self.area.uuid)
+        self._connected = ExternalStrategyConnectionManager.register(
+            self._redis_communicator, self.channel_prefix,
+            self.connected, self._get_transaction_id(payload),
+            area_uuid=self.area.uuid)
 
     def _unregister(self, payload):
-        self._connected = unregister_area(self._redis_communicator, self.channel_prefix,
-                                          self.connected, self._get_transaction_id(payload))
+        self._connected = ExternalStrategyConnectionManager.unregister(
```

Comment:
Shouldn't this be
```suggestion
        self._is_registered = ExternalStrategyConnectionManager.unregister(
```
now?

Reply:
@hannesdiedrich Actually no, I was not refactoring the `redi_external_market_connection`, I only updated this function name in the refactoring
So this flag is not touched here (I only renamed the flag in the external strategies)

Weirdly though, I took a look at the `redi_external_market_connection` _connected flag just now, and it seems that it's not used at all (we just set it to True/False but we do not reference it anywhere)
I suppose this is not harmful at the moment but should indeed be refactored in a separate PR, do you agree?

Reply:
I see, this is a bug. The register and unregister functionality does not work for the market connection. The `self.connected` flag of the area is never checked in any of the market commands.
`_register` always worked before because `self.connected` is always False.
And apparently nobody ever called `_unregister` which would have revealed the error. We probably only used the aggregator connection lately
Please open a bug story actually. It is not crucial, as for the REST connection via d3a-web we have another layer of security, but still, the direct connection should be working correctly even though nobody is using it. Thanks!

Reply:
Sounds good, I have created the ticket!
https://gridsingularity.atlassian.net/browse/D3ASIM-3678